markdown_lint/linter.py

```
@@ -0,0 +1,317 @@
+
+    def _check_heading(self, report: FileReport, line_num: int, line: str, match: re.Match) -> None:
+        """Check heading formatting and spacing."""
+        level = len(match.group("level"))
+        content = match.group("content")
+
+        # Check for space after heading markers
+        if not line.startswith(f"{'#' * level} "):
+            self._add_issue(
+                report,
+                line_num,
+                f"Missing space after heading marker",
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)